internal/expr/query.go

```diff
@@ -8,36 +8,38 @@ import (
 	"strings"
 )
 
-func (qe *QueryExpr) QueryArgs() []any {
-	return qe.args
+// PrimedQuery contains all concrete values needed to run a SQLair query on a
+// database.
+type PrimedQuery struct {
+	sql string
+	// params are the query parameters to pass to the database.
+	params []any
+	// outputs specifies where to scan the query results.
+	outputs []typeMember
 }
 
-func (qe *QueryExpr) HasOutputs() bool {
-	return len(qe.outputs) > 0
+// Params returns the query parameters to pass with the SQL to a database.
+func (pq *PrimedQuery) Params() []any {
+	return pq.params
 }
 
-type QueryExpr struct {
-	sql     string
-	args    []any
-	outputs []typeMember
+// HasOutputs returns true if the SQLair query contains at least one output
+// expression.
+func (pq *PrimedQuery) HasOutputs() bool {
+	return len(pq.outputs) > 0
 }
 
-// Query returns a query expression ready for execution, using the provided values to
-// substitute the input placeholders in the prepared expression. These placeholders use
-// the syntax "$Person.fullname", where Person would be a type such as:
-//
-//	type Person struct {
-//	        Name string `db:"fullname"`
-//	}
-func (pe *PreparedExpr) Query(args ...any) (ce *QueryExpr, err error) {
+// BindInputs takes the SQLair input arguments and returns the PrimedQuery ready
+// for use with the database.
+func (te *TypedExpr) BindInputs(args ...any) (pq *PrimedQuery, err error) {
```

Comment:
Move this to be co-located with its type declaration in _typebind.go_

Reply:
Moved to `typebind.go`.

Following this pattern the `func (*ParsedExpr) BindTypes(...)` method in `typebind.go` should really be moved to the parser to be colocated with the declaration of `ParsedExpr`. To me though that move would feel a bit wrong, all the actual "type binding" takes place in that method so it should really be in `bindtypes.go`. I have left it as it is for now.